### ISSUE TITLE
Ban Warning!

### ISSUE THREAD

**Issue by Emuuung** [jabbink/PokemonGoBot#1110]
Ok so I got this warning message after (apparently) geting a new TOS agreement, the client is not detected as the original:  **Your account may be banned! please try from the official client.**
PS: this is not a new account btw.
```
C:\PokemonGoBot>gradlew run
Download https://jitpack.io/com/github/Grover-c13/PokeGOAPI-Java/Development-SNAPSHOT/PokeGOAPI-Java-Development-0.2-gc678019-228.jar
:compileKotlin
w: C:\PokemonGoBot\src\main\kotlin\ink\abb\pogo\scraper\util\pokemon\PokemonData.kt: (25, 41): Parameter 'playerlevel' is never used
:compileJava
:processResources UP-TO-DATE
:createProperties
:classes
:run
                                  ,'\
    _.----.        ____         ,'  _\   ___    ___     ____        ;-.               ,
_,-'       `.     |    |  /`.   \,-'    |   \  /   |   |    \  |`.   \ '.           .'/
\      __    \    '-.  | /   `.  ___    |    \/    |   '-.   \ |  |   \  \ .---. .-' /
 \.    \ \   |  __  |  |/    ,','_  `.  |          | __  |    \|  |    '. '     `\_.'
   \    \/   /,' _`.|      ,' / / / /   |          ,' _`.|     |  |      |(),()  |     ,
    \     ,-'/  /   \    ,'   | \/ / ,`.|         /  /   \  |     |      (  __   /   .' \
     \    \ |   \_/  |   `-.  \    `'  /|  |    ||   \_/  | |\    |     .''.___.'--,/\_,|
      \    \ \      /       `-.`.___,-' |  |\  /| \      /  | |   |    {  /     \   }   |
       \    \ `.__,'|  |`-._    `|      |__| \/ |  `.__,'|  | |   |     '.\     /_.'    /
        \_.-'       |__|    `-._ |              '-.|     '-.| |   |      |'-.-',  `; _.'
                                `'                            '-._|      |  |  |   |`
                                                                         `""`""`"""`
16 ago 00:11:06 [default] - Trying to read C:\PokemonGoBot\config.properties
       16 ago 00:11:07 [default] - Logging in to game server...
       16 ago 00:11:12 [default] - Logged in successfully
Getting profile data from pogo server.
16 ago 00:11:13 [default] - Accepting ToS
       Exception in thread "default" com.pokegoapi.exceptions.AsyncPokemonGoException: Unknown exception occurred.
        at com.pokegoapi.util.AsyncHelper.toBlocking(AsyncHelper.java:45)
        at com.pokegoapi.main.RequestHandler.sendServerRequests(RequestHandler.java:155)
        at ink.abb.pogo.scraper.MainKt.startBot(Main.kt:259)
        at ink.abb.pogo.scraper.MainKt.startDefaultBot(Main.kt:148)
        at ink.abb.pogo.scraper.PokemonGoBotApplication$BotRunner$run$1.invoke(PokemonGoBotApplication.kt:58)
        at ink.abb.pogo.scraper.PokemonGoBotApplication$BotRunner$run$1.invoke(PokemonGoBotApplication.kt:47)
        at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:18)
Caused by: java.lang.RuntimeException: java.util.concurrent.ExecutionException: com.pokegoapi.exceptions.RemoteServerException: Your account may be banned! please try from the official client.
        at rx.exceptions.Exceptions.propagate(Exceptions.java:58)
        at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:465)
        at rx.observables.BlockingObservable.first(BlockingObservable.java:168)
        at com.pokegoapi.util.AsyncHelper.toBlocking(AsyncHelper.java:37)
        ... 6 more
Caused by: java.util.concurrent.ExecutionException: com.pokegoapi.exceptions.RemoteServerException: Your account may be banned! please try from the official client.
        at com.pokegoapi.main.RequestHandler$1.get(RequestHandler.java:109)
        at com.pokegoapi.main.RequestHandler$1.get(RequestHandler.java:86)
        at rx.internal.operators.OnSubscribeToObservableFuture$ToObservableFuture.call(OnSubscribeToObservableFuture.java:74)
        at rx.internal.operators.OnSubscribeToObservableFuture$ToObservableFuture.call(OnSubscribeToObservableFuture.java:43)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.subscribe(Observable.java:9957)
        at rx.Observable.subscribe(Observable.java:9924)
        at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:445)
        ... 8 more
Caused by: com.pokegoapi.exceptions.RemoteServerException: Your account may be banned! please try from the official client.
        at com.pokegoapi.main.RequestHandler.internalSendServerRequests(RequestHandler.java:223)
        at com.pokegoapi.main.RequestHandler.run(RequestHandler.java:293)
        at java.lang.Thread.run(Thread.java:745)
> Building 83% > :run```

**Comment by jabbink** [comment]
Hi @Emuuung,

It looks like there is some information missing from your ticket that will be needed in order to diagnose and fix the problem at hand. Please take a look at the [Contribution Guidelines](https://github.com/jabbink/PokemonGoBot/blob/develop/.github/CONTRIBUTING.md) and [default Issue template (the one you did not follow)](https://github.com/jabbink/PokemonGoBot/blob/develop/.github/ISSUE_TEMPLATE.md), which will tell you **exactly** what your ticket has to contain in order to be processable.

I'm marking this one now as needing some more information. Please understand that if you do not provide that information within the next two days (until 2016-08-18 07:20) I'll close this ticket so it doesn't clutter the bug tracker.

Best regards,
~ Your friendly GitIssueBot

PS: I'm just an automated script, not a human being.

**Comment by patzor** [comment]
maybe your IP is banned?

**Comment by masiarb** [comment]
@patzor I am getting the same error but running multiple accounts, and the other one is running just fine so I guess it's not an IP ban.

**Comment by AlexBoulyga** [comment]
Maybe your ACC is banned?

**Comment by staurakis** [comment]
I got the exact same situation. Now the weird part is how the ban came.
Yesterday night i logged in with the Niantic app. It was ok. Then i used the bot, everything was also ok and the bot run for about an hour and then i stopped it.
When i tried to run it again this morning (few hours later) i also received the above message which for me means that the actual ban came the moment the bot run. In other words, the have a way to identify the bot the moment it logs in and before it do anything and the ban it....
Just for the history, now the Niantic client displays the message: 'cannot retrieve account data from server'. So it is a ban for i don't know how long.
I thought it would be good to share these with you.....

**Comment by gielie** [comment]
After i got my 2 newly created accounts banned yesterday i tried a new account this morning. 
After about 1,5 hour the account got banned, with all settings set on slow and took available precautions.

I think they have a way now to detect this bot (i dont know about other bots)

**Comment by staurakis** [comment]
Yes i am sure about it as well... They have a way to immediately detect it and ban the account.

**Comment by dvasko** [comment]
didn't run bot for 5 days and still got permaBan

**Comment by AlexBoulyga** [comment]
My bot is running and no van, are you guys shure? Maybe they just check the IP and immidiately van any acc from it?

**Comment by Danbardo** [comment]
same

**Comment by AlexBoulyga** [comment]
As far as I can say, i guess there is this new ToS Service accepting API call, and if you dont do it correctly/manually it recognizes you are a bot

**Comment by MacDogg** [comment]
I have a lvl 8 account I used with Necrobot a week or so ago.  I only used it for a few hours before I ran it on my main.  So, I started using that account, via proxy, right now with this bot.  I'll see how long it goes or see if it gets banned in a little while.
I used a new account with this bot since Saturday and stopped it last night before bed and haven't used it since so I am unsure if that account is working at the moment but will try it in BlueStacks in a little bit and see.

**Comment by AlexBoulyga** [comment]
@MacDogg Plesse use good settings for randomization too

**Comment by HoreaM** [comment]
@staurakis The exact same thing happened to me, 2 minutes ago.

**Comment by MacDogg** [comment]
Well, my bot went from level 8.5 to 13 and I was able to quit the bot and then login via bluestacks with no error.  I do use all of the randomizers in the profiles file.

**Comment by HoreaM** [comment]
@MacDogg I checked the account before using the bot, it wasn't banned. It got banned when the bot was starting up, in the exact same spot as the Emuuung posted.

**Comment by AlexBoulyga** [comment]
@HoreaM and I said it might be this ToS thing that you must manually agree to

**Comment by KyleBoyer** [comment]
> Ok so I got this warning message after (apparently) geting a new TOS agreement, the client is not detected as the original: Your account may be banned! please try from the official client.
> PS: this is not a new account btw.

Google or PTC account?

The things that occurred prior to that stack trace (I believe) are:

0.(If you are using PTC) Login thru PTC account
0.(If you are using Google) Login thru Google account
1. Get profile data
2. Accept TOS

So that means, the issue could be in getting profile data or accepting terms of service. However, if it is a PTC account, the issue could also be when logging into the server which raised a flag.

**Comment by MacDogg** [comment]
I never got a ToS agree window before or after botting this morning.

**Comment by staurakis** [comment]
I have accepted the tos throught the cluent and run the bot correctly afterwards. And the without any obvious reason i got banned the moment the bot tried to run again.

**Comment by jmp00000001** [comment]
You guys use Android or IPhone?
The bot (like all others) fakes an IPhone if I remember correctly.

**Comment by MacDogg** [comment]
I log in with BlueStacks to check the status.
Currently getting the Pokestop out of range error with two different proxy addresses entered.  I did login via BlueStacks and it seems to work ok in there, I was able to spin a stop and get the rewards.
Odd some people are getting banned right after logging in.

**Comment by derekbaalmann** [comment]
Running 4 bots, 1 got banned last week, 2 yesterday/last night. I have one more that's working but I stopped running it for now since it seems semi-insta-ban-ish. 
Of the 3 that got banned, the first was a PTC, second and third (banned at the same time) are ptc and google. The working one is google.
Always stayed up to date with releases (I stopped running from source awhile ago).
I tried to make it as random/humanlike as possible, then copied the config to all 4 accounts (obviously updating the config per release). 
So either it's just odd/random on getting banned or it's some sort of flag system. You get flagged as a bot (maybe once or more than once? maybe through the api?) then they ban in a wave. The one account that still works is the most recent one, so the other 3 could have been flagged for past behavior.
I'm going to sit back and try logging in with that last account tomorrow and see what it does, I'm 90% it's been flagged and just waiting in the bunch to get banned.

Thanks for all the great work, probably going to stop bothering with this game. My wife and I have had more fun configuring and running bots, scrapers, mappers, GPS spoofing, general tools and the like more than we ever had playing a broken game (granted played WAY more when the step counter worked and played a bit more when android mappers were available but....)

**Comment by jmp00000001** [comment]
https://github.com/OpenPoGo/OpenPoGoBot/issues/319

could also be the unclaimed lvl reward

**Comment by derekbaalmann** [comment]
Tried to login today on the third and final bot through the app on a Nexus 6. Banned. Wasn't banned yesterday, I used the phone app twice yesterday to confirm it wasn't banned. Didn't run any bots for the last 48 hours or any other api, so it was definitely flagged, then banned. FYI. Also we used these bot accounts a lot on our phones so while I can't confirm that the first two didn't have unclaimed rewards, I know the third was watched more carefully and never level'ed without us logining in for rewards. The third bot was only lvl 24, while the other 2 were close to lvl 30 or past (I didn't watch those as closely).
Just putting the info out there for others trying to figure this out.
